### PR TITLE
Add experiment state indicators to commit records in table

### PR DESCRIPTION
**Pull request by mattseddon**
# 1/3 #3585 <- this <- #3595 <- #3596

Look a "new" feature.

This PR moves the experiment indicators (count) from experiment rows up to commit records. Previously we only did this for checkpoints. That no longer makes sense. 

I also fixed a bug with the top-level plotted indicator. Hidden rows were being incorrectly removed from the count. 

### Demo

https://user-images.githubusercontent.com/37993418/228984127-3d3e4b88-fda0-4eb3-b2c7-644bd1d7b2bb.mov

### CODE REVIEW

File: extension/src/experiments/webview/contract.ts [hunk 1]
```diff
@@ -103,9 +105,8 @@ export type TableData = {
   hasValidDvcYaml: boolean
   isShowingMoreCommits: boolean
   rows: Commit[]
+  selectedForPlotsCount: number
```
Comment (mattseddon) [inline extension/src/experiments/webview/contract.ts:108]: [F] Easier/more consistent to pass this from the extension.

File: webview/src/experiments/components/App.test.tsx [hunk 2]
```diff
@@ -249,6 +253,83 @@ describe('App', () => {
     })
   })
 
+  describe('Sub-rows middle states indicators', () => {
```
Comment (mattseddon) [inline webview/src/experiments/components/App.test.tsx:256]: [F] These were deleted previously. I have adapted the old tests to work on the commit record.

File: webview/src/experiments/util/rows.ts [hunk 3]
```diff
@@ -1,19 +0,0 @@
-import { Experiment } from 'dvc/src/experiments/webview/contract'
-import { Row } from '@tanstack/react-table'
-
-const addToSelected = (
-  selectedForPlotsCount: number,
-  row: Row<Experiment>
-): number => selectedForPlotsCount + (row.original?.selected ? 1 : 0)
-
-export const getSelectedForPlotsCount = (
```
Comment (mattseddon) [inline webview/src/experiments/util/rows.ts:9]: [F] I cannot explain why I did this... maybe responsiveness... 😕

File: webview/src/stories/Table.stories.tsx [hunk 4]
```diff
@@ -130,6 +136,28 @@ WithSurvivalData.args = {
...
+const tableDataWithSomeSelectedExperiments = setExperimentsAsSelected(
+  tableData,
+  ['4fb124a', '42b8736', '1ba7bcd']
+)
+WithMiddleStates.args = {
+  tableData: setExperimentsAsStarred(tableDataWithSomeSelectedExperiments, [
+    '1ba7bcd'
+  ])
+}
+WithMiddleStates.play = async ({ canvasElement }) => {
```
Comment (mattseddon) [inline webview/src/stories/Table.stories.tsx:149]: [F] Again this story was deleted but now I've brought it back

File: webview/src/test/sort.ts [hunk 5]
```diff
@@ -63,6 +63,7 @@ export const tableData: TableData = {
       label: 'main'
     }
   ],
+  selectedForPlotsCount: 0,
   sorts: []
```
Comment (mattseddon) [inline webview/src/test/sort.ts:67]: [Q] Why is this file even here???? Shouldn't it be with the fixtures 😕

File: webview/src/experiments/components/App.test.tsx [hunk 6]
```diff
@@ -1205,6 +1288,43 @@ describe('App', () => {
       )
     })
 
+    it('should show not change the plotted indicator when plotted experiments are hidden', () => {
```
Comment (mattseddon) [inline webview/src/experiments/components/App.test.tsx:1291]: [F] This is a new test that would have caught the bug.

File: webview/src/experiments/components/table/CellRowActions.tsx [hunk 7]
```diff
@@ -52,10 +49,7 @@ export const CellRowAction: React.FC<CellRowActionProps> = ({
 
   return (
     <CellHintTooltip tooltipContent={tooltipContent}>
-      <div
-        className={cx(styles.rowActions, hidden && styles.hidden)}
-        data-testid={testId}
-      >
+      <div className={cx(styles.rowActions)} data-testid={testId}>
```
Comment (mattseddon) [inline webview/src/experiments/components/table/CellRowActions.tsx:52]: [F] I tried to clean up the classes and props but this needs a good clean out.

File: webview/src/experiments/components/table/Indicators.tsx [hunk 8]
```diff
@@ -43,7 +43,7 @@ export const Indicator = ({
 }) => {
   const content = (
     <button
-      className={cx(styles.indicatorIcon, count && styles.indicatorWithCount)}
```
Comment (mattseddon) [inline webview/src/experiments/components/table/Indicators.tsx:46]: `indicatorWithCount` also no longer existed

File: webview/src/experiments/components/table/styles.module.scss [hunk 9]
```diff
@@ -566,7 +567,7 @@ $bullet-size: calc(var(--design-unit) * 4px);
           align-items: center;
           justify-content: center;
           width: 1.3rem;
-          height: 100%;
+          height: 2rem;
```
Comment (mattseddon) [inline webview/src/experiments/components/table/styles.module.scss:570]: [F] Indicator was in the middle of the parent icon

File: webview/src/experiments/components/table/styles.module.scss [hunk 10]
```diff
@@ -974,16 +975,11 @@ $badge-size: 0.85rem;
...
-      display: none;
       .experimentGroup & {
-        display: block;
         &[aria-label='0'] {
           display: none;
         }
       }
-      .experimentGroup.expandedGroup & {
-        display: none;
-      }
```
Comment (mattseddon) [inline webview/src/experiments/components/table/styles.module.scss:986]: just 😕

**Review by sroy3 — APPROVED** [review]

File: webview/src/experiments/components/table/Indicators.tsx (hunk 8 above)
Comment (mattseddon) [reply webview/src/experiments/components/table/Indicators.tsx:46]: I dumb

**Pull request merged**